### PR TITLE
[Dashboard] Change key for hosts from hostname to (hostname, ip)

### LINKED ISSUE #9157: [dashboard] 0.8.6 two nodes with same name cause flickering
I have two Ubuntu 19.10 with the exact same config (0.8.6) and name. Just the IP is different (I cloned a VM).
Then in the dashboard it shows only one of both. It switches between both (IP is changing).

Start head:
ray start --head --port=6379

worker:
ray start --address='192.168.40.129:6379' --redis-password='5241590000000000'

As soon as I changed the name of the second vm, it works as expected.

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
Multiple nodes may have the same IP address and different hostnames, and in addition nodes may have the same hostname and different IP addresses.  In our `node_stats` dictionary, then, we should use the tuple `(hostname, ip)` as the key instead of just `hostname`.

## Related issue number

Closes https://github.com/ray-project/ray/issues/9157.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested (please justify below)
I don't know how to automatically start up a cluster with specified hostnames and IP addresses.  However, we tested this on an AWS cluster and manually changed the hostname of one of the worker nodes to match the head node, and the dashboard successfully displayed each one in its own row, rather than flickering between them as before.

### CODE REVIEW

File: python/ray/dashboard/dashboard.py [hunk 1]
```diff
@@ -333,15 +333,15 @@ async def kill_actor(self, req) -> aiohttp.web.Response:
...
-        result = self.dashboard_controller.get_logs(hostname, pid)
+        result = self.dashboard_controller.get_logs(ip, pid)
         return await json_response(self.is_dev, result=result)
 
     async def errors(self, req) -> aiohttp.web.Response:
-        hostname = req.query.get("hostname")
+        ip = req.query.get("ip_address")
         pid = req.query.get("pid")
-        result = self.dashboard_controller.get_errors(hostname, pid)
+        result = self.dashboard_controller.get_errors(ip, pid)
```
Comment (mfitton) [inline python/ray/dashboard/dashboard.py:344]: These calls have corresponding API requests in the front-end that will also need to be changed that exist in `api.ts`.

**Review by mfitton — CHANGES_REQUESTED** [review]
I'm pretty sure this won't work as-is, since the front-end has a couple of requests that will need to be updated to use ip instead of hostname. Also, it seems like the data returned to the front-end is now different (the keys of the `clients` in the node stats response is a tuple instead of just the hostname), and I think there are some front-end adjustments there. 

To be honest, fixing this ends up being substantially more complicated than I'd originally thought by the looks of things, given this new PR from Ant to achieve roughly the same thing (although a bit different since it uses Node IDs): https://github.com/ray-project/ray/pull/10528#pullrequestreview-487662915

I'm wondering if it's worth continuing on this because I think getting it out for Ray 1.0 will take some additional effort, and I'm working on moving our front-end over to the new backend anyway. 

Interested in what @edoakes thinks about that.

File: python/ray/dashboard/dashboard.py (hunk 1 above)
Comment (architkulkarni) [reply python/ray/dashboard/dashboard.py:344]: I see, that makes sense.  Your concerns about continuing make sense; I'm happy with whatever you and/or @edoakes decide.

**Comment by edoakes** [comment]
Yeah no need to duplicate it, let's just push the new backend through.

**Pull request closed without merging**